Source/CarthageKit/Project.swift

```diff
@@ -1036,7 +1007,29 @@ public final class Project { // swiftlint:disable:this type_body_length
 				}
 			}
 			.flatMap(.concat) { dependencies -> SignalProducer<(Dependency, PinnedVersion), CarthageError> in
-				return .init(dependencies)
+				return SignalProducer(dependencies)
+					.flatMap(.concurrent(limit: 4)) { dependency, version -> SignalProducer<(Dependency, PinnedVersion), CarthageError> in
+						switch dependency {
+						case .git, .gitHub:
+							guard options.useBinaries else {
+								return .empty
+							}
+							return self.installBinaries(for: dependency, pinnedVersion: version, toolchain: options.toolchain)
+								.filterMap { installed -> (Dependency, PinnedVersion)? in
+									return installed ? (dependency, version) : nil
+							}
+						case let .binary(url):
+							return self.installBinariesForBinaryProject(url: url, pinnedVersion: version, projectName: dependency.name, toolchain: options.toolchain)
+								.then(.init(value: (dependency, version)))
+						}
+					}
+					.collect()
+					.map { installedDependencies -> [(Dependency, PinnedVersion)] in
+						return dependencies.filter { dependency -> Bool in
+							!installedDependencies.contains { $0 == dependency }
+						}
```

Comment:
This should be able to be `return dependencies.filter { !installedDependencies.contains($0) }`

Reply:
The type of `installedDependencies` is `[(Dependency, PinnedVersion)]` and `dependency` is `(Dependency, PinnedVersion)` here so cannot use `.contains(_ element: Element)` here because Element is not technically equatable.